### PR TITLE
PrefixQueryParser takes a String as value like its Builder

### PR DESCRIPTION
**Pull request by alexksikes**
Relates #12032

### CODE REVIEW

**Comment by alexksikes** [comment]
@jpountz can you please have a look at this?

File: core/src/main/java/org/elasticsearch/index/query/PrefixQueryParser.java [hunk 1]
```diff
@@ -73,7 +73,7 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
                         if ("_name".equals(currentFieldName)) {
                             queryName = parser.text();
                         } else if ("value".equals(currentFieldName) || "prefix".equals(currentFieldName)) {
-                            value = parser.objectBytes();
+                            value = parser.textOrNull();
```
Comment (javanna) [inline core/src/main/java/org/elasticsearch/index/query/PrefixQueryParser.java]: same as in the regex PR, I wonder if we should have parser.text() or parser.textOrNull()

Comment (alexksikes) [inline core/src/main/java/org/elasticsearch/index/query/PrefixQueryParser.java]: reasoning here is that we check for null at the end and return a probably more comprehensive error message.

Comment (jpountz) [inline core/src/main/java/org/elasticsearch/index/query/PrefixQueryParser.java]: makes sense to me

**Comment by jpountz** [comment]
LGTM

**Pull request closed without merging**